applications/GeoMechanicsApplication/custom_conditions/U_Pw_face_load_interface_condition.cpp

```diff
@@ -190,26 +190,24 @@ void UPwFaceLoadInterfaceCondition<3,4>::CheckJointWidth(double& rJointWidth, bo
                                                             const double& MinimumJointWidth, const Element::GeometryType& Geom)
 {
     //Quadrilateral_interface_3d_4
-    array_1d<double, 3> pmid0;
-    array_1d<double, 3> pmid1;
-    array_1d<double,3> P2 = Geom.GetPoint( 2 );
-    noalias(pmid0) = 0.5 * (Geom.GetPoint( 0 ) + Geom.GetPoint( 3 ));
-    noalias(pmid1) = 0.5 * (Geom.GetPoint( 1 ) + P2);
+    const auto& P2 = Geom.GetPoint( 2 );
+    const auto& pmid0 = 0.5 * (Geom.GetPoint( 0 ) + Geom.GetPoint( 3 ));
+    const auto& pmid1 = 0.5 * (Geom.GetPoint( 1 ) + P2);
     
     //Unitary vector in local x direction
-    array_1d<double,3> Vx;
-    noalias(Vx) = pmid1 - pmid0;
-    double inv_norm_x = 1.0/norm_2(Vx);
+    const auto& tmpVx = pmid1 - pmid0;
+    array_1d<double,3> Vx{tmpVx[0], tmpVx[1], tmpVx[2]};
+    const auto inv_norm_x = 1.0/norm_2(Vx);
```

Comment:
```suggestion
    const double inv_norm_x = 1.0/norm_2(Vx);
```
We avoid using `auto` for basic types. Maybe it is also a good idea to check that `Vx` is non-zero here.

Reply:
Can you please explain what is the rationale behind not using `auto` for basic types (I presume that with "basic types" you refer to built-in types like `int`, `bool`, etc). When it comes to `auto` I tend to follow the [AAA-style](https://herbsutter.com/2013/08/12/gotw-94-solution-aaa-style-almost-always-auto/) for the reasons set forward by Herb Sutter.

Reply:
I think some people got burnt by using `auto` with expression templates (`auto` would bind to an expression instead of forcing an evaluation, potentially causing performance issues later down the line due to unwanted lazy evaluation) and decided never to use it again.

As for built-ins, I'm not sure. I can imagine a scenario where one would unintentionally mix signed and unsigned indices/IDs that could potentially cause over/underflow, but
1) that's not the case here
2) the compiler warns about these cases so we'd be better off treating these warnings as errors.

Reply:
Besides @matekelemen observation, at some point we took into account more "practical" reasons such as avoid accidentally using a "wrong" type (float / int divisions and things of this sort) and code readability.